cmd/thanos/receive.go

```diff
@@ -84,6 +84,8 @@ func registerReceive(m map[string]setupFunc, app *kingpin.Application) {
 
 	forwardTimeout := modelDuration(cmd.Flag("receive-forward-timeout", "Timeout for forward requests.").Default("5s").Hidden())
 
+	uploadTimeout := modelDuration(cmd.Flag("receive.upload-timeout", "Timeout for the initial and last upload request.").Default("10m").Hidden())
```

Comment:
It's very tricky if for initial and last. Why not all? (:

Reply:
Yep, we can use for all for sure.